sklearn/feature_selection/_rfe.py

```diff
@@ -729,6 +738,15 @@ def fit(self, X, y, groups=None):
 
         # Build an RFE object, which will evaluate and score each possible
         # feature count, down to self.min_features_to_select
+        n_features = X.shape[1]
+        if self.min_features_to_select > n_features:
+            warnings.warn(
+                (
+                    f"Found {self.min_features_to_select=} > {n_features=}. There will"
+                    " be no feature selection and all features will be kept."
+                ),
+                UserWarning,
+            )
         rfe = RFE(
             estimator=self.estimator,
             n_features_to_select=self.min_features_to_select,
```

Comment:
We can pass a curated `n_features_to_select` so that we're sure the warning won't be raised again in the inner RFE.
```suggestion
            n_features_to_select=min(self.min_features_to_select, n_features),
```